File: nodejs/local.js
const fs = require("node:fs");

const stat = fs.statSync("test_file.txt");
console.log("File size (in bytes): " + stat.size);
console.log("Last modified time: " + stat.mtime.toUTCString());

const fd = fs.openSync("test_file.txt", "rs+");
const buf = Buffer.alloc(200);
let status = fs.readSync(fd, buf, 0, 5);
console.log(status + " " + buf);
// Adding the 5th argument to readSync is equivalent to calling
// lseek(fd, 0, SEEK_SET) beforehand
status = fs.readSync(fd, buf, 5, 6, 0);
console.log(status + " " + buf);
fs.writeSync(fd, " word"); // need leading space for some reason
fs.ftruncateSync(fd, 30);
fs.fsyncSync(fd);
fs.closeSync(fd);

fs.renameSync("test_file.txt", "renamed_file.txt");
fs.unlinkSync("deleted_file.txt");

fs.readdirSync(".", {withFileTypes: true}).forEach((entry) => {
    if (entry.isDirectory()) {
        console.log(entry.name + "/");
    }
    if (entry.isFile()) {
        console.log(entry.name);
    }
});

// To use Unix domain sockets:
// Server: https://nodejs.org/api/net.html#serverlistenpath-backlog-callback
// Client: https://nodejs.org/api/net.html#netcreateconnectionpath-connectlistener

File: csharp/Local.cs
FileInfo f = new("test_file.txt");
Console.WriteLine("File size (in bytes): " + f.Length);
Console.WriteLine("Last modified time: " + f.LastWriteTimeUtc.ToString("R"));

FileStream file = new("test_file.txt", FileMode.OpenOrCreate, FileAccess.ReadWrite);
byte[] buf = new byte[200];
int status = file.Read(buf, 0, 5);
Console.WriteLine(status + " " + System.Text.Encoding.UTF8.GetString(buf));
file.Seek(0, SeekOrigin.Begin);
status = file.Read(buf, 5, 6);
Console.WriteLine(status + " " + System.Text.Encoding.UTF8.GetString(buf));
file.Write(System.Text.Encoding.UTF8.GetBytes("word"), 0, 4);
file.SetLength(30);
file.Flush();
file.Close();

f.MoveTo("renamed_file.txt");
File.Delete("deleted_file.txt");

var namelist = Directory.EnumerateFileSystemEntries(".");
foreach (var name in namelist)
{
    FileAttributes attributes = File.GetAttributes(name);
    if ((attributes & FileAttributes.Directory) == FileAttributes.Directory)
    {
        Console.WriteLine(name[2..] + "/");
    }
    if ((attributes & FileAttributes.Normal) == FileAttributes.Normal)
    {
        Console.WriteLine(name[2..]);
    }
}
